### DISCUSSION TITLE
Why "NotifyCanExecuteChangedFor" didn't fire?

### DISCUSSION THREAD

**Discussion by Phil4121** [CommunityToolkit/dotnet#552] [Q&A] [answered]
Hello!

Currently I'm playing around with the CommunityToolkit for .Net MAUI and DevExpress Controls for .Net MAUI.

In my LoginViewModel I'm using the following code snippets:

I've a class for saving mailadress and password of the user like this:

    public partial class LoginModel : ObservableObject
    {
        [ObservableProperty]
        private string email;

        [ObservableProperty]
        private string password;
    }

In my ViewModel I've got a property of this class which also manage the execution of a command:

    [ObservableProperty]
    [NotifyCanExecuteChangedFor(nameof(LoginCommand))]
    private LoginModel loginCredentials = new();

And now the implementation of the command:

    [RelayCommand(CanExecute = nameof(ValiateLoginData))]
    private async Task LoginAsync()
    {
        try
        {
            IsBusy = true;

            Guard.ArgumentNotNull(LoginCredentials, nameof(LoginCredentials));

            var user = await Auth.AuthenticateUserAsync(LoginCredentials.Email, LoginCredentials.Password);
            await TryUserSignInAsync(user, tryOnlineAuthentication: false);
        }
        finally
        {
            IsBusy = false;
        }
    }

ValidateLoginData checks the user input:

    private bool ValiateLoginData()
    {
        return  !string.IsNullOrEmpty(LoginCredentials.Email) &&
                !string.IsNullOrEmpty(LoginCredentials.Password)
    }

Mail and passwort are inserted by user via GUI and are stored in Email and Passwort Property of LoginModel but [NotifyCanExecuteChangedFor] isn't called at all!

Could someone please get me some help?

Thanks
Phil

**Comment by pictos** [comment]
cc: @Sergio0694

**Comment by Sergio0694** [comment, marked as answer]
This is by design. `[NotifyCanExecuteChangeFor]` will notify the command _when the annotated property (field) changes_. In your case that's not what's happening: you're changing a property inside an object that's assigned to the property. But the value of the property itself is not changing — it's still the same object.

Unrelated: you don't need that `IsBusy` boilerplate code and additional property within the command, just bind to the `IsRunning` property of the generated async relay command, that'll do all of that automatically for you 🙂

**Reply by Phil4121** [reply]
Thank you Sergio for your quick reply!
